lib/backends/sdc/containers.js

```diff
@@ -559,60 +525,302 @@ function addRulesToFWAPI(opts, rules, payload, callback) {
     });
 }
 
-
 /**
- * Add networks to the payload: 'external' if no fabrics are enabled.  If
- * fabrics are enabled, then add the user's default fabric network.  If
- * fabrics are enabled and the container is publishing ports, also add the
- * NAT pool as a public-facing network.
+ * Add networks to the payload. Dispatches to one of these supported networking
+ * configurations:
+ *   1. without fabrics configured: external networking only.
+ *
+ *   With fabrics enabled:
+ *   2. 'bridge' or nothing specified provisions on the user's default network,
+ *      potentially also with the external network.
+ *   3. Specifying a network name will provision to the named *fabric* network.
+ *   4. Specifying a network id (or portion) will provision to that *fabric*
+ *      network.
+ *
+ *   Docker resolves name/id collisions in the following way:
+ *     - a name is preferred to a partial id
+ *     - a full id is preferred to a name
  */
 function addNetworksToPayload(opts, container, payload, callback) {
     assert.object(opts, 'opts');
     assert.object(opts.config, 'opts.config');
     assert.object(opts.config.napi, 'opts.config.napi');
     assert.object(opts.log, 'opts.log');
+
+    assert.object(opts.config.overlay);
```

Comment:
Does `assert-plus` need a second string parameter to produce a meaningful assertion message, like with the assert statements above?